kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -147,9 +147,14 @@ public SnapshotImpl getSnapshotForTimestamp(
   public static void verifyDeltaVersionsContiguous(List<Long> versions, Path tablePath) {
     for (int i = 1; i < versions.size(); i++) {
       if (versions.get(i) != versions.get(i - 1) + 1) {
+        long expectedVersion = versions.get(i - 1) + 1;
+        long actualVersion = versions.get(i);
         throw new InvalidTableException(
             tablePath.toString(),
-            String.format("Missing delta files: versions are not contiguous: (%s)", versions));
+            String.format(
+                "Missing delta files: versions are not contiguous. Expected version %d after "
+                    + "version %d, but found version %d. All versions: %s",
+                expectedVersion, versions.get(i - 1), actualVersion, versions));
```

Comment:
I'm not sure if this change is particularly helpful given we already include the full list of versions? (which will evidently be non-contiguous). Are you thinking this will just be easier to parse than the whole thing?